README.md

```diff
@@ -1,49 +1,73 @@
 To spin up things in an OpenShift cluster based on the contents in this
-repository, you can follow any of the two methods of getting openshift cluster
+repository, please make sure you have a minishift based VM or a CentOS VM with
+root privileges. You'll also need to spin up Docker Distribution (registry) on
+same VM or different VM.
 
-***Minishift:***
+###Docker Distribution (registry) setup
 
-To have a
-[minishift](https://github.com/minishift/minishift/) based VM started up,we used following command to start a minishift environment:
+The system on which you'd like to setup the registry, execute following
+commands:
 
 ```bash
-$ minishift start --disk-size 50GB --memory 8GB --iso-url centos --openshift-version 3.9.0
+$ yum install -y docker-distribution
+$ systemctl enable --now docker-distribution
 ```
 
-But the resources can be varied based on availability. However, make sure to
-use `--iso-url centos` part in above command as we have setup things on CentOS
-based minishift VM.
+Also make sure that the firewall rules are not blocking access to the registry
+(port 5000 by default.)
 
-***CentOs Based VM:***
+###OpenShift setup
 
-Get a CentOS based VM with minimum 4GB RAM, 50GB Diskspace
+**Minishift**
 
-Then install docker and enable openshift origin repos
+Start the minishift VM using below command:
 
 ```bash
-yum install docker git centos-release-openshift-origin -y
-yum install origin-clients -y
+$ minishift start --disk-size 50GB --memory 8GB --iso-url centos --openshift-version 3.9.0 --insecure-registry <registry-ip>:<port>
 ```
-Edit Docker config to support openshifts internal registry. Update `/etc/docker/daemon.json`
 
+Memory and storage can be varied based on availability. It is recommended to
+have 4GB memory and 20GB disk space as minimum. However, make sure to use
+`--iso-url centos` part in above command as we have setup things on CentOS based
+minishift VM. 
+
+**CentOS VM**
+
+A CentOS VM with 8GB memory and 50GB disk space should suffice. You can adjust
+the resources based on availability. It is recommended to have 4GB memory and
+20GB disk space as minimum.
+
+In the VM, install docker and enable openshift origin repos:
+
+```bash
+$ yum install -y docker git centos-release-openshift-origin
+$ yum install -y origin-clients
 ```
+
+Edit Docker config to support OpenShift's internal registry and the external
+registry we created in earlier step. Update `/etc/docker/daemon.json`
+
+```json
 {
-"insecure-registries":["172.30.0.0/16"]
+"insecure-registries":["172.30.0.0/16", "<registry-ip>:<port>"]
```

Comment:
we are making this insecure registry only for push right?

Reply:
Yes.